accounts-db/src/accounts.rs

```diff
@@ -552,31 +569,52 @@ impl Accounts {
         account_locks: &mut AccountLocks,
         writable_keys: Vec<&Pubkey>,
         readonly_keys: Vec<&Pubkey>,
-    ) -> Result<()> {
+        entry_accts_lookup: &mut EntryAcctLocks,
+        allow_self_conflicting_entries: bool,
+    ) -> (Result<()>, bool) {
+        let mut self_conflicting_account: bool = false;
+        let mut self_conflicting_tx = false;
         for k in writable_keys.iter() {
             if account_locks.is_locked_write(k) || account_locks.is_locked_readonly(k) {
-                debug!("Writable account in use: {:?}", k);
-                return Err(TransactionError::AccountInUse);
+                if allow_self_conflicting_entries {
+                    self_conflicting_account = entry_accts_lookup.writables.contains(k)
+                        || entry_accts_lookup.readables.contains(k);
+                }
+                if !self_conflicting_account {
+                    debug!("Writable account in use: {:?}", k);
+                    return (Err(TransactionError::AccountInUse), self_conflicting_tx);
+                } else {
+                    self_conflicting_tx = true;
```

Comment:
self_conflicting_tx can be set to `true` if `entry_accts_lookup.writables.contains(k) == false` and `entry_accts_lookup.readables.contains(k) == false`.

is that semantically correct?

Reply:
no, you are right, this is not semantically correct.

although, `self_conflicting_tx` will have no effect if the batch locking fails on the replay side in `process_entries` (which is part of the next PR). 
To be clearer, I will return false for every error condition Err(TransactionError::AccountInUse)